### ISSUE TITLE
Form FieldArray keep reference to old items value when adding new record to the array.

### ISSUE THREAD

**Issue by simonssspirit** [telerik/kendo-react#650]
Form FieldArray keeps a reference to old items value when adding a new record to the array.

This can be observed in our demo if we change the field to use our CheckBox component:

https://stackblitz.com/edit/react-5xte1s?file=app/main.jsx

Steps
1) Add a new item and check it.
2) Add a second item and observe how the checkbox is added as checked even if the value is false.

**Comment by Steven-Chorkawy** [comment]
Hello @simonssspirit, 

Has there been any update on this issue?

**Comment by simonssspirit** [comment]
After more research we found why this occurs:

- with Checkbox - This is an issue inside the Checkbox component and it is logged here - https://github.com/telerik/kendo-react/issues/699

We just released a new feature that allows passing name in the props to allow unique key generation inside the Checkbox field for a workaround:

https://stackblitz.com/edit/react-5xte1s-swqmb4?file=app/main.jsx

- with standard DOM inputs, the issue is fixed and it is already in the dev channel- https://github.com/telerik/kendo-react/issues/688

- with Switch, the Switch is working with checked instead of value and a wrapper is needed. The reason is that the Switch uses a value to set not only a boolean value. This is why the checked is required:  https://stackblitz.com/edit/react-5xte1s-qs3ixa?file=app/main.jsx

As this is ended up not caused by the Form component, this issue will be closed.
